backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/lowlevel/index/settings/impl/IndexSettings.java

```diff
@@ -90,6 +102,13 @@ public void merge(IndexSettings other) {
 			maxResultWindow = other.maxResultWindow;
 		}
 
+		if ( knn == null ) {
+			knn = other.knn;
+		}
+		else {
+			knn = knn || Boolean.TRUE.equals( other.knn );
```

Comment:
Javadoc says:

```
Any conflict of definition will be solved in favor of {@code this}.
```

I think it's important to allow users to override Hibernate Search settings.

Reply:
good point! 😃 thanks